src/bridges/notional/NotionalBridge.sol

```diff
@@ -0,0 +1,228 @@
+// SPDX-License-Identifier: GPL-2.0-only
+pragma solidity >=0.8.4;
+
+import {ERC20} from "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import {AztecTypes} from "../../aztec/libraries/AztecTypes.sol";
+import {ErrorLib} from "../base/ErrorLib.sol";
+import {BridgeBase} from "../base/BridgeBase.sol";
+import {IWrappedfCashFactory} from "../../interfaces/notional/IWrappedfCashFactory.sol";
+import {IWrappedfCash} from "../../interfaces/notional/IWrappedfCash.sol";
+import {IRollupProcessor} from "../../aztec/interfaces/IRollupProcessor.sol";
+import {WETH9} from "../../interfaces/notional/WETH9.sol";
+import {CToken} from "../../interfaces/notional/CToken.sol";
+
+contract NotionalBridgeContract is BridgeBase {
+    IWrappedfCashFactory public fcashFactory;
+    address public constant ETH = address(0);
+    address public constant CETH = address(0x4Ddc2D193948926D02f9B1fE9e1daa0718270ED5);
+    address public constant DAI = address(0x6B175474E89094C44Da98b954EedeAC495271d0F);
+    address public constant CDAI = address(0x5d3a536E4D6DbD6114cc1Ead35777bAB948E3643);
+    address public constant WETH = address(0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2);
+    address public constant USDC = address(0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48);
+    address public constant CUSDC = address(0x39AA39c021dfbaE8faC545936693aC917d5E7563);
+    address public constant WBTC = address(0x2260FAC5E5542a773Aa44fBCfeDf7C193bc2C599);
+    address public constant CWBTC = address(0xC11b1268C1A384e55C48c2391d8d480264A3A7F4);
+    mapping(address => uint16) public currencyIds;
+    mapping(address => bool) public ctokens;
+    mapping(address => address) public underlyingAddrMap;
+
+    constructor(address _rollupProcessor, address _fcashFactory) BridgeBase(_rollupProcessor) {
+        fcashFactory = IWrappedfCashFactory(_fcashFactory);
+        currencyIds[ETH] = 1;
+        currencyIds[CETH] = 1;
+        currencyIds[WETH] = 1;
+        currencyIds[DAI] = 2;
+        currencyIds[CDAI] = 2;
+        currencyIds[USDC] = 3;
+        currencyIds[CUSDC] = 3;
+        currencyIds[WBTC] = 4;
+        currencyIds[CWBTC] = 4;
+        underlyingAddrMap[CETH] = WETH;
+        underlyingAddrMap[CDAI] = DAI;
+        underlyingAddrMap[CUSDC] = USDC;
+        underlyingAddrMap[CWBTC] = WBTC;
+        ctokens[CETH] = true;
+        ctokens[CDAI] = true;
+        ctokens[CUSDC] = true;
+        ctokens[CWBTC] = true;
```

Comment:
Setting currencyIds in constructor is not a good idea because it can happen that Notional adds support for another cToken and then the bridge could not support that. Doesn't notional have a contract similar to Compound's Comptroller which tracks all the supported markets? Check our [Compound bridge](https://github.com/AztecProtocol/aztec-connect-bridges/blob/a233b8ebd5e8648c100c86de430532bffc3576cf/src/bridges/compound/CompoundBridge.sol#L53) for how we handle it there.